### ISSUE TITLE
Word2Vec visualization error 

### ISSUE THREAD

**Issue by vin101** [deeplearning4j/deeplearning4j#1798]
I am following the documentation and can create the vectors but the visualisation is not working either with TSNE or Barnes Hut Tsne. The program executes but the TSNE tab on the localhost says " no t-SNE data available"

public class V1 {
    private static org.slf4j.Logger log = LoggerFactory.getLogger(Word2VecRawTextExample.class);

    public static void main(String [] args) {

        log.info("Load data....");
        SentenceIterator iter = new LineSentenceIterator(new File("/Users/vinayak/Desktop/twitter4j-4.0.4/Tay/vectors/Vec1/input.txt"));
        iter.setPreProcessor(new SentencePreProcessor() {
            @Override
            public String preProcess(String sentence) {
                return sentence.toLowerCase();
            }
        });

        log.info("Tokenize data....");
        final EndingPreProcessor preProcessor = new EndingPreProcessor();
        TokenizerFactory tokenizer = new DefaultTokenizerFactory();
        tokenizer.setTokenPreProcessor(new TokenPreProcess() {
            @Override
            public String preProcess(String token) {
                token = token.toLowerCase();
                String base = preProcessor.preProcess(token);
                base = base.replaceAll("\\d", "d");
                if (base.endsWith("ly") || base.endsWith("ing"))
                    System.out.println();
                return base;
            }
        });

        int batchSize = 1000;
        int iterations = 3;
        int layerSize = 150;

        log.info("Build model....");
        Word2Vec vec = new Word2Vec.Builder()
                .batchSize(batchSize) //# words per minibatch.
                .minWordFrequency(5) //
                .useAdaGrad(false) //
                .layerSize(layerSize) // word feature vector size
                .iterations(iterations) // # iterations to train
                .learningRate(0.025) //
                .minLearningRate(1e-3) // learning rate decays wrt # words. floor learning
                .negativeSample(10) // sample size 10 words
                .iterate(iter) //
                .tokenizerFactory(tokenizer)
                .build();
        vec.fit();

        try {
            WordVectorSerializer.writeWordVectors(vec, "words.txt");
        } catch (IOException e) {
            e.printStackTrace();
        }

        log.info("Plot TSNE....");
        Tsne t1= new Tsne.Builder()
                .setMaxIter(1000)
                .stopLyingIteration(250)
                .learningRate(500)
                .useAdaGrad(false)
                .setMomentum(0.5)
                .normalize(true)
                .usePca(false)
                .build();

        /**BarnesHutTsne tsne = new BarnesHutTsne.Builder()
                .setMaxIter(1000)
                .stopLyingIteration(250)
                .learningRate(500)
                .useAdaGrad(false)
                .theta(0.5)
                .setMomentum(0.5)
                .normalize(true)
                .usePca(false)
                .build(); **/
        //vec.lookupTable().plotVocab(tsne);
        UiServer server = null;
        try {
            server = UiServer.getInstance();
        } catch (Exception e) {
            e.printStackTrace();
        }
        System.out.println("Started on port " + server.getPort());
       vec.lookupTable().plotVocab(t1, 1000, server.getConnectionInfo());

    }
}

**Comment by Goldado** [comment]
Could you attach the words.txt file please?

**Comment by vin101** [comment]
[input.txt](https://github.com/deeplearning4j/deeplearning4j/files/355118/input.txt)
[words.txt](https://github.com/deeplearning4j/deeplearning4j/files/355119/words.txt)
@Goldado  Here is the input file and the word.txt, sorry for the delay

**Comment by Goldado** [comment]
@vin101 There doesn't seem to be a problem with the net's output. I came just this afternoon across an error when I first opened UiServer on Chrome, I had the "no data ... please upload it as txt or retry later" smthn of that sort (seems to be the one you had). I just kept reopening localhost:8080 and clicking TSNE, and it worked about the third time. Give that a try.

**Comment by vin101** [comment]
@Goldado Thanks it worked! Well it works sometimes. Refreshing multiple time solved it, but it is nondeterministic. It works sometimes and sometimes you are stuck with the following error:

No t-SNE data available! Please upload something, or check again later.×

**Comment by raver119** [comment]
That sounds pretty strange, and pretty wrong. As far as i see, this thing is more then determenistic: once T-SNE caclulations are finished - coordinates are posted to UiServer, and that's the end of the process :) Once it's available on server - all subsequent requests for those coords are served from there.

So, question is: are you able to use network monitor built into Chrome? Maybe that's just some caching issue?

**Comment by vin101** [comment]
@raver119 Yes once the graph is visualized, consecutive attempts are successful. The problem is loading the graph in the first place. Sometimes it loads with 5-6 browser refreshes and sometimes it doesn't and I have to re-run the entire process.

I have tried using Safari to isolate the issue, but the same problem persists.

**Comment by MehmetAbdulhay** [comment]
Hi,

I have been facing the same issue for a while and I could not find the solution so far. Is there any improvement so far? Here is my related code part:

        UiServer server = UiServer.getInstance();
        System.out.println("Started on port " + server.getPort());

        log.info("Plot TSNE....");
        BarnesHutTsne tsne = new BarnesHutTsne.Builder()
            .setMaxIter(5)
            .stopLyingIteration(5)
            .learningRate(500)
            .useAdaGrad(false)
            .theta(0.5)
            .setMomentum(0.5)
            .normalize(true)
            //.usePca(false)
            .build();


        vec.lookupTable().plotVocab(tsne, 1000, server.getConnectionInfo());

When I run this piece of code the results will be as below:

INFO  [2016-10-03 15:07:35,355] org.deeplearning4j.plot.BarnesHutTsne: Handled 0 records
INFO  [2016-10-03 15:07:39,148] org.deeplearning4j.plot.BarnesHutTsne: Handled 500 records
INFO  [2016-10-03 15:07:54,360] org.deeplearning4j.plot.BarnesHutTsne: Error at iteration 0 is -2983099.094312263
INFO  [2016-10-03 15:08:03,633] org.deeplearning4j.plot.BarnesHutTsne: Error at iteration 1 is -2983101.751996019
INFO  [2016-10-03 15:08:17,323] org.deeplearning4j.plot.BarnesHutTsne: Error at iteration 2 is -2983102.0065930625
INFO  [2016-10-03 15:08:25,033] org.deeplearning4j.plot.BarnesHutTsne: Error at iteration 3 is -2983084.0188392065
INFO  [2016-10-03 15:08:33,003] org.deeplearning4j.plot.BarnesHutTsne: Error at iteration 4 is -2982997.5690521626

Even though it throws no exception, when I connect to http://localhost:8080/ and click to T-SNE, it says "No T-SNE data available..."

How should I proceed?

Thanks in advance.

**Comment by raver119** [comment]
T-SNE is special case, it shows result only after calculations are finished, so it's not a real-time Listener.

**Comment by MehmetAbdulhay** [comment]
It has been working fro more than 12 hours for a data set which is not normal I guess. The output still shows INFO [2016-10-03 15:08:33,003] org.deeplearning4j.plot.BarnesHutTsne: Error at iteration 4 is -2982997.5690521626 as its last line. Any idea?

Thanks.

**Comment by raver119** [comment]
That depends on amount of input data, data dimensionality, number of iterations, etc. So no, this doesn't sound like something impossible.
